hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -153,9 +153,12 @@ private void checkIdClass(
 						return;
 					}
 				}
+				else if ( idClass.isAssignableFrom( event.getEntityId().getClass() ) ) {
```

Comment:
This is the fix, right? IMO we should stick to the `isInstance` check if possible instead of using `isAssignableFrom`

Reply:
Thanks @beikov , changed it